src/vos/vos_obj_cache.c

```diff
@@ -254,6 +254,9 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 
 	if (obj->obj_df) {
 		D_DEBUG(DB_TRACE, "looking up object ilog");
+		if (create || intent == DAOS_INTENT_PUNCH)
+			vos_ilog_ts_ignore(vos_obj2umm(obj),
```

Comment:
vos_ilog_ts_add() will directly return if vos_ts_in_tx() is false, under such case, do we still need to call vos_ilog_ts_ignore()? On the other hand, I would suggest to move vos_ilog_ts_ignore() into related ilog functions, such as vos_ilog_ts_add()/vos_ilog_ts_mark(). Otherwise, new added callers may miss to call vos_ilog_ts_ignore().

Reply:
This patch gets us to clean results on vos tests so at least for that case, we don't need to call it if vos_ts_in_tx is false.

I considered moving into ilog functions but I need the umm and it's not available in those functions without changing them.   Can I defer cleanup of this interface to a future patch?

Reply:
It's OK for me.